test/parallel/test-http-response-drain-cork.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+const common = require('../common');
+const http = require('http');
+const assert = require('assert');
+
+// Test that drain event is emitted correctly when using cork/uncork
+// with ServerResponse and the write buffer is full
+
+const server = http.createServer(common.mustCall(async (req, res) => {
+  res.cork();
+  
+  // Write small amount - won't need drain
+  assert.strictEqual(res.write('1'.repeat(100)), true);
+  
+  // Write large amount that should require drain
+  const needsDrain = !res.write('2'.repeat(1000000));
+  
+  if (needsDrain) {
```

Comment:
This should always need drain right? It's corked so there's no race condition or anything.

Unless there's a good reason, imo it would be better to enforce that (`assert.strictEqual(res.write('2'.repeat(1000000)), false)`) and drop the `if` here entirely.